raphtory-graphql/src/model/graph/graph.rs

```diff
@@ -1,111 +1,250 @@
-use std::{
-    collections::{HashMap, HashSet},
-    ops::Deref,
-};
-
 use crate::model::{
     algorithms::graph_algorithms::GraphAlgorithms,
     filters::{edge_filter::EdgeFilter, node_filter::NodeFilter},
-    graph::{edge::Edge, get_expanded_edges, node::Node, property::Property},
+    graph::{edge::Edge, get_expanded_edges, node::Node, property::GqlProperties},
     schema::graph_schema::GraphSchema,
 };
 use dynamic_graphql::{ResolvedObject, ResolvedObjectFields};
 use itertools::Itertools;
 use raphtory::{
+    core::entities::vertices::vertex_ref::VertexRef,
     db::{
-        api::view::{DynamicGraph, IntoDynamic, StaticGraphViewOps, TimeOps, VertexViewOps},
+        api::{
+            properties::dyn_props::DynProperties,
+            view::{DynamicGraph, TimeOps, VertexViewOps},
+        },
         graph::edge::EdgeView,
     },
     prelude::*,
     search::IndexedGraph,
 };
+use std::{
+    collections::{HashMap, HashSet},
+    convert::Into,
+    ops::Deref,
+};
 
 #[derive(ResolvedObject)]
-pub(crate) struct GraphMeta {
+pub(crate) struct GqlGraph {
     name: String,
-    graph: DynamicGraph,
+    graph: IndexedGraph<DynamicGraph>,
 }
 
-impl GraphMeta {
-    pub fn new(name: String, graph: DynamicGraph) -> Self {
+impl GqlGraph {
+    pub fn new(name: String, graph: IndexedGraph<DynamicGraph>) -> Self {
         Self { name, graph }
     }
 }
 
 #[ResolvedObjectFields]
-impl GraphMeta {
-    async fn name(&self) -> String {
-        self.name.clone()
+impl GqlGraph {
+    ////////////////////////
+    // LAYERS AND WINDOWS //
+    ////////////////////////
+
+    async fn unique_layers(&self) -> Vec<String> {
+        self.graph.unique_layers().map_into().collect()
     }
 
-    async fn static_properties(&self) -> Vec<Property> {
-        self.graph
-            .properties()
-            .constant()
-            .into_iter()
-            .map(|(k, v)| Property::new(k.into(), v))
-            .collect()
+    async fn default_layer(&self) -> GqlGraph {
+        GqlGraph::new(
+            self.name().await,
+            self.graph.default_layer().into_dynamic_indexed(),
+        )
     }
 
-    async fn node_names(&self) -> Vec<String> {
+    async fn layers(&self, names: Vec<String>) -> Option<GqlGraph> {
+        let name = self.name().await;
         self.graph
-            .vertices()
-            .into_iter()
-            .map(|v| v.name())
-            .collect_vec()
+            .layer(names)
+            .map(move |g| GqlGraph::new(name, g.into_dynamic_indexed()))
```

Comment:
If we pull out `into_dynamic_indexed` into a trait, we could make `new` here generic and not have to worry about calling `into_dynamic_indexed` everywhere explicitly.